File: lib/data/models/food/food_allergen_model.dart
class FoodAllergenModel {
  const FoodAllergenModel({this.id, this.name, this.value});

  factory FoodAllergenModel.fromJson(Map<String, dynamic> json) {
    return FoodAllergenModel(
      id: json['id'] as String?,
      name: json['name'] as String?,
      value: json['value'] as bool?,
    );
  }

  final String? id;
  final String? name;
  final bool? value;
}

File: server/src/models/foods/FoodAllergenModel.js
class FoodAllergenModel {
  constructor({ id, name, value }) {
    this.id = id;
    this.name = name;
    this.value = value;
  }

  static fromJson(json) {
    return new FoodAllergenModel({
      id: json["allergen_id"],
      name: json["allergen_name"],
      value: json["allergen_value"] == "1" ? true : json["allergen_value"] == "0" ? false : null,
    });
  }
}

export default FoodAllergenModel;
